### PR TITLE
Nixify the build

### LINKED ISSUE #4: Pin Kube version?
Currently we get the swagger spec from the `master` branch of Kubernetes.

Now, Kube should not remove any specifications, but they do deprecate some. On our end everything should be fine, as we never remove any types/default, we only add them.

Should we pin the Kube version? If yes, should we support multiple versions?

### LINKED ISSUE #30: Create a nix derivation that stores generated files in nix store
This way, I can depend on `dhall-kubernetes`'s output in my  kubernetes project easily without
having to generate the files myself.

### LINKED ISSUE #41: Cannot build with Nix sandboxing
Using 
```
nix-build release.nix -A dhall-kubernetes --arg src '{rev = "";}'
```

Fails because `dhall-1.18` fails to build. `dhall-1.18` test suite makes HTTP requests, which is verboten in Nix, causing the test suite to fail

### PR DESCRIPTION
**Pull request by arianvp**
Fixes #41, fixes #4 and fixes #30 

Currently uses whatever k8s version nixpkgs packages, but in my experience it is kept rather up to date.  

We might want to make an `./update-nixpkgs.sh` script that quickly updates to the latests `nixpkgs-unstable` whenever we want to bump it

### CODE REVIEW

File: release.nix
Comment (f-f) [inline release.nix]: I think we need this for Hydra to always report builds?

**Review by f-f — APPROVED** [review]
Amazing 👏 💯 🎉

**Comment by f-f** [comment]
@arianvp +1 on the `./update-nixpkgs.sh`. Can we actually pin to a precise Kube version? Should we care?

File: release.nix
Comment (arianvp) [reply release.nix]: Ah good point.

**Comment by arianvp** [comment]
Well it is pinned to a specific Nixpkgs version, which also pins to a specific k8s version. I guess that is good enough (for now).  The actual `dhall-kubernetes.nix` file is parameterised over the Kubernetes pkgs, so we could use this to later easily build types for multiple k8s versions if that is something people want. However we'll have to think about how we distribute the Dhall files to the user then. I'm still not sure if putting them directly GitHub repo is the best approach here but I don't have any other good ideas

**Comment by f-f** [comment]
@arianvp I think GitHub is pretty great, because you can pin to specific commits (and I hope everyone does, since we'll have some good amount of breaking changes with #46) or tags (we don't do releases, but we'll start. I think it's a good idea to have a release just before we merge the Haskell rewrite).

If there's demand for building types for multiple k8s versions, then we have several possibilities while staying on GitHub: several folders, or several branches, etc

**Comment by arianvp** [comment]
Issues should be addressed, and there's now a single script `./generate.sh` that will both generate `types` and `default` and will generate `README.md`

**Comment by arianvp** [comment]
I added an option to automatically update nixpkgs now @f-f I think that is all and this is ready to merge!

**Comment by f-f** [comment]
@arianvp wonderful! This looks great 🙂
I'll let you to merge it so you can write a commit message for the squashed commits

**Pull request merged**
